maintainers/scripts/update-cuda-packages.sh

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env bash
+# -----------------------------------------------------------------------------
+# DESRCIPTION:
+#   Updates NVIDIA CUDA packages (`cudaPackages`) from cuda-redist and runfile.
+#
+# AUTHOR:
+#   Adam Erickson (Nervosys)
+#
+# DATE:
+#   2024-05-01
+#
+# OPTIONAL:
+#   - Updated supported GPUs here: pkgs/development/cuda-modules/gpus.nix
+#
+# NOTES:
+#   - Assumes CUDA_VERSION_MAJOR_MINOR_PREVIOUS was a minor version decrement
+#   - For documentation, see `nixpkgs/doc/languages-frameworks/cuda_section.md`
+# -----------------------------------------------------------------------------
+
+set -e                      # exit on error
+[ "$DEBUG" = 1 ] && set -x  # print program trace
+
+###
+### Variables
+###
+
+# Static: defaults
+CUDA_VERSION='12.4.1'
+CUTENSOR_VERSION='2.0.1'
+CUDA_RUNFILE_URL='https://developer.download.nvidia.com/compute/cuda/12.4.1/local_installers/cuda_12.4.1_550.54.15_linux.run'
+
+###
+### Functions
+###
+
+function usage {
+    printf "\nUsage: %s [-c|--cuda] CUDA_VERSION [-t|--tensor] CUTENSOR_VERSION\n" $0
+}
+
+###
+### Main
+###
+
+# Parse command-line interface (CLI) arguments.
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    -c|--cuda)
+      CUDA_VERSION="$2"
+      shift  # past argument
+      shift  # past value
+      ;;
+    -t|--cutensor)
+      CUTENSOR_VERSION="$2"
+      shift  # past argument
+      shift  # past value
+      ;;
+    -u|--cuda-runfile-url)
+      CUDA_RUNFILE_URL="$2"
+      shift  # past argument
+      shift  # past value
+      ;;
+    -*|--*)
+      echo "Error: unknown argument: $1"
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+# Dynamic variables
+CUDA_VERSION_MAJOR_MINOR="${CUDA_VERSION%.*}"
+CUDA_VERSION_MAJOR_MINOR_UNDER="${CUDA_VERSION_MAJOR_MINOR/./_}"
+CUDA_VERSION_MAJOR_MINOR_PREVIOUS="$(echo "scale=10; ${CUDA_VERSION_MAJOR_MINOR} - 0.1" | bc -l)"
+CUDA_VERSION_MAJOR_MINOR_PREVIOUS_UNDER="${CUDA_VERSION_MAJOR_MINOR_PREVIOUS/./_}"
+
+echo '-----------------------------------------------'
+echo " CUDA version new:      ${CUDA_VERSION} || ${CUDA_VERSION_MAJOR_MINOR} || ${CUDA_VERSION_MAJOR_MINOR_UNDER}"
+echo " CUDA version previous:        || ${CUDA_VERSION_MAJOR_MINOR_PREVIOUS} || ${CUDA_VERSION_MAJOR_MINOR_PREVIOUS_UNDER}"
+echo '-----------------------------------------------'
+
+### Download and process manifests.
+
+# Update CUDA: download and process manifests.
+nix run github:connorbaker/cuda-redist-find-features -- \
+    download-manifests \
+    --log-level DEBUG \
+    --version "$CUDA_VERSION" \
+    https://developer.download.nvidia.com/compute/cuda/redist \
+    ./pkgs/development/cuda-modules/cuda/manifests
```

Comment:
I'd say we should package `cuda-redist-find-features` in Nixpkgs (like a normal `src = fetchFromGitHub` package) and use it in the update script accordingly. This would also allow us to remove flake references from the Nixpkgs manual (they currently don't belong there)